maven/WFLY-14934_Server_Plugin.adoc

```diff
@@ -0,0 +1,208 @@
+// Remove the x next to the relevant field if the feature in question is not relevant
+// to that kind of WildFly installation
+* [ ] Traditional standalone server (unzipped or provisioned by Galleon)
+
+* [ ] Managed domain
+
+* [x] OpenShift s2i
+
+* [x] Maven-controlled standalone server
+
+* [ ] Bootable jar
+
+== Context
+
+Although this feature is not limited to the cloud context, the cloud execution context is the main driver for this feature.
+
+Today, in order to execute WildFly in an Openshift context, we are relying on WildFly S2I (Source-to-Image) builder and runtime images to build 
+and execute an WildFly application image.
+There is a strong coupling inside this image between the server version and the docker image that implies 
+releasing a new image for each new WildFly release. This approach is not flexible. We should be able to build 
+an application image for any WildFly release (even for a locally built SNAPSHOT) without having to wait for the 
+WildFly builder image to be built/deployed. 
+
+In addition, the WildFly server located inside the builder image, enforces a set of configuration API (a set of environment variables)
+and bash launch scripts. At startup time, a complex CLI script generation that impacts the server startup time is mandatorily executed.
+Having this level of configurability should be a user choice and not be enforced by the WildFly s2i image.
+
+To solve these pain points, we are removing in WFLY-14936 the server installation and its configuration aspects from the image.
+
+The WildFly S2I image becomes a generic image allowing the installation and execution of any provisioned WildFly server.
+
+Due to this removal, we need a way for customers to provision the WildFly server to be installed inside the new image 
+when building their application. This WildFly server will run the built application inside the image.
+
+To do so we are defining Maven tooling that handles the Galleon provisioning of the server, the application deployment and the configuration aspects.
+
+== Requirements
+
+The existing https://github.com/wildfly/wildfly-maven-plugin/[WildFly Maven plugin] has been designed to interact (mainly) with a running server instance. 
+We are here proposing to evolve the capabilities of this plugin for one to be able to build a fully configured server containing application deployment.
+ 
+This proposal adds a new kind of installation that is operated and controlled by the pom.xml file.  The pom.xml file contains all the information needed
+to control the server and application content and versioning.
+
+=== Hard Requirements
+
+==== Plugin requirements
+
+* The Maven plugin produces inside the project target dir a directory (named ```server``` by default) that contains the server and any deployment.
+
+==== WildFly Maven Plugin new goals
+
+* ```wildfly:package```: Galleon provisioning of a server, CLI script execution, copy content to the server, deploy application (if present). 
```

Comment:
I'm not sure if we want to put that in this proposal but the key aspect for our new S2I images is that we will rely on this `package` goal only.
It makes sense to introduce a `provision` goal to split the provisioning from the packaging of WildFly but that's not something that will be called by S2I.

Reply:
Ok, done.